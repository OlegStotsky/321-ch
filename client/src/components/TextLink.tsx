import * as React from "react";
import { Link } from "react-router-dom";

interface ITextLinkProps {
  children: React.ReactNode;
  href: string;
  color?: string;
  classNames?: string[];
}

const TextLink: React.SFC<ITextLinkProps> = ({
  children,
  href,
  color,
  classNames
}) => {
  return (
    <Link
      className={`text-link text-link--${color} ${
        classNames ? classNames.join(" ") : null
      }`}
      to={href}
    >
      {children}
    </Link>
  );
};

export default TextLink;
