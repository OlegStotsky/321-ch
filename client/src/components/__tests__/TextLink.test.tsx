import * as React from "react";
import * as Enzyme from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import TextLink from "../TextLink";

Enzyme.configure({ adapter: new Adapter() });

describe("TextLink", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = Enzyme.shallow(<TextLink data="abcd" href="/" />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("has a with text-link className", () => {
    expect(wrapper.find(".text-link")).toHaveLength(1);
  });
  it("provides data correctly", () => {
    expect(wrapper.contains("abcd")).toBe(true);
  });
  it("provides href into a element correctly", () => {
    expect(wrapper.find({ href: "/" })).toHaveLength(1);
  });
});
