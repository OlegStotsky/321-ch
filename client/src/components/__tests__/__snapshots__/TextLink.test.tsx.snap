// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextLink
    href="/"
  >
    abcd
  </TextLink>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "abcd",
      "className": "text-link text-link--undefined null",
      "replace": false,
      "to": "/",
    },
    "ref": null,
    "rendered": "abcd",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "abcd",
        "className": "text-link text-link--undefined null",
        "replace": false,
        "to": "/",
      },
      "ref": null,
      "rendered": "abcd",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
